pkg/controller/master/setting/nodeconfig.go

```diff
@@ -0,0 +1,156 @@
+package setting
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	nodev1 "github.com/harvester/node-manager/pkg/apis/node.harvesterhci.io/v1beta1"
+	longhorntypes "github.com/longhorn/longhorn-manager/types"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	ctlnode "github.com/harvester/harvester/pkg/controller/master/node"
+	harvSettings "github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+// syncNodeConfig handles changes to "longhorn-v2-data-engine-enabled" and
+// "ntp-servers" settings, as both of these need to be pushed out to all nodes
+// via each node's nodeconfig CR.
+func (h *Handler) syncNodeConfig(setting *harvesterv1.Setting) error {
+	logrus.WithFields(logrus.Fields{
+		"name":  setting.Name,
+		"value": setting.Value,
+	}).Info("Processing setting")
+
+	nodes, err := h.nodeClient.List(metav1.ListOptions{})
+	if err != nil {
+		logrus.WithError(err).Error("Failed to list all nodes")
+		return err
+	}
+
+	// Enqueue all the nodes so that nodeOnChanged can sync the node config.
+	for _, node := range nodes.Items {
```

Comment:
Two questions here:
(1) Should we enqueue only after the successful processing of the following code? or always wakeup the nodes.

(2) From the context, it looks LH and each node takes time to enable/disable the V2 engine; how to prevent a user from enable & disable it on UI frequently? thanks.

Reply:
1. We have to enqueue the nodes first, because the subsequent setting of `lhs/v2-data-engine=true` won't succeed until all the nodes are in the correct state (i.e. have enough hugepages allocated). It may take a little while for the nodes to be in the right state, but the LH setting change will be requeued and it will stick in due course.
2. I hadn't considered a means of preventing users from enabling/disabling it frequently. As this is a preview feature I was planning to go into some detail in the documentation about the need to apply the setting, then wait for it to take effect, and possibly reboot nodes if necessary (see also the text in the GUI PR at https://github.com/harvester/dashboard/pull/1108)

Reply:
Definitely, when it is not perfect, we can document it and enhance later.